ebb_lint/flake8.py

```diff
@@ -247,8 +252,18 @@ def _message_for_pos(self, pos, error, **kw):
         return lineno, column, message, type(self)
 
     def run(self):
-        with open(self.filename, 'r') as infile:
-            self.future_features = detect_future_features(infile)
+        if self.filename == 'stdin':
+            try:
+                enc = getattr(sys.stdin, 'encoding') or 'utf-8'
+                stdin_reader = codecs.getreader(enc)(sys.stdin)
+                self.future_features = detect_future_features(stdin_reader)
+            finally:
+                # like the `with open()` pattern, tries to ensure we can read
+                # from this source again in the future.
+                sys.stdin.seek(0)
+        else:
+            with open(self.filename, 'r') as infile:
+                self.future_features = detect_future_features(infile)
```

Comment:
Aw jeez. Is this how flake8 indicates that one is supposed to read from stdin? I hadn't looked into it yet.